packages/types/src/types.d.ts

```diff
@@ -27,12 +28,43 @@ export type SnapRpcHandler = (args: {
 
 export type OnRpcRequestHandler = SnapRpcHandler;
 
-export type SnapProvider = MetaMaskInpageProvider;
+export type TransactionInsight = {
+  insights: { [key: string]: unknown };
+  cancel?: boolean;
```

Comment:
The `cancel` was an example, I'm not sure we want to support in production for now, that would require a new UI flow for it as well.